cardano-api/src/Cardano/Api/ProtocolParametersUpdate.hs

```diff
@@ -0,0 +1,674 @@
+                      , perUnitTime :: Lovelace
+                      } deriving (Eq, Show)
+
+_fromPrices :: Alonzo.Prices -> ExecutionUnitPrices
+_fromPrices (Alonzo.Prices pMem pStep) =
+  ExecutionUnitPrices (fromShelleyLovelace pMem) (fromShelleyLovelace pStep)
+
+_toPrices :: ExecutionUnitPrices -> Alonzo.Prices
+_toPrices (ExecutionUnitPrices pMem pStep) =
+  Alonzo.Prices (toShelleyLovelace pMem) (toShelleyLovelace pStep)
+
+instance FromJSON ExecutionUnitPrices where
+  parseJSON = withObject "ExecutionUnitPrices" $ \o -> do
+    obj <- o .: "executionUnitPrices"
+    ExecutionUnitPrices <$> obj .: "unitSpace" <*> obj .: "unitTime"
+
+instance ToJSON ExecutionUnitPrices where
+  toJSON (ExecutionUnitPrices perSpace perTime) =
+    object [ "executionUnitPrices" .= object
+             [ "unitSpace" .= perSpace , "unitTime" .= perTime]
+           ]
+
+newtype MaxTxExecutionUnits =
+    MaxTxExecutionUnits { unMaxTxExecutionUnits :: ExecutionUnits}
+    deriving (Eq, Show)
+
+_fromMaxTxExec :: Alonzo.ExUnits -> MaxTxExecutionUnits
+_fromMaxTxExec (Alonzo.ExUnits mMem mSteps) =
+  MaxTxExecutionUnits $ ExecutionUnits mMem mSteps
+
+_toTxExecUnits :: MaxTxExecutionUnits -> Alonzo.ExUnits
+_toTxExecUnits (MaxTxExecutionUnits (ExecutionUnits mMem mSteps)) =
+  Alonzo.ExUnits mMem mSteps
+
+instance ToJSON MaxTxExecutionUnits where
+  toJSON (MaxTxExecutionUnits (ExecutionUnits space time)) =
+    object [ "maxTxExecutionUnits" .=
+                object ["space" .= space, "time" .= time]
+           ]
+
+instance FromJSON MaxTxExecutionUnits where
+  parseJSON = withObject "MaxTxExecutionUnits" $ \o -> do
+    obj <- o .: "maxTxExecutionUnits"
+    MaxTxExecutionUnits
+      <$> (ExecutionUnits <$> obj .: "space" <*> obj .: "time")
+
+newtype MaxBlockExecutionUnits =
+    MaxBlockExecutionUnits { unMaxBlockExecutionUnits :: ExecutionUnits}
+    deriving (Eq, Show)
+
+_fromMaxBlockExec :: Alonzo.ExUnits -> MaxBlockExecutionUnits
+_fromMaxBlockExec (Alonzo.ExUnits mMem mSteps) =
+  MaxBlockExecutionUnits $ ExecutionUnits mMem mSteps
+
+_toExUnits :: MaxBlockExecutionUnits -> Alonzo.ExUnits
+_toExUnits (MaxBlockExecutionUnits (ExecutionUnits mMem mSteps)) =
+  Alonzo.ExUnits mMem mSteps
+
+instance ToJSON MaxBlockExecutionUnits where
+  toJSON (MaxBlockExecutionUnits (ExecutionUnits space time)) =
+    object [ "maxBlockExecutionUnits" .=
+                object ["space" .= space, "time" .= time]
+           ]
+
+instance FromJSON MaxBlockExecutionUnits where
+  parseJSON = withObject "MaxBlockExecutionUnits" $ \o -> do
+    obj <- o .: "maxBlockExecutionUnits"
+    MaxBlockExecutionUnits
+      <$> (ExecutionUnits <$> obj .: "space" <*> obj .: "time")
+
+
+data ExecutionUnits
+    = ExecutionUnits { space :: Word64
+                     , time :: Word64
+                     } deriving (Eq, Show)
+
+
+
+-- ----------------------------------------------------------------------------
+-- Conversion functions
+--
```

Comment:
I can't tell what we've changed here since we've split the module. Can I suggest that's another good reason to not split them in this PR, or at least not split them and change them in a single commit.